tiledb/sm/crypto/encryption_key.h

```diff
@@ -56,6 +57,15 @@ class EncryptionKey {
   /** Destructor. This zeroes the underlying key buffer. */
   ~EncryptionKey();
 
+  /**
+   * Constructor initializing object utilizing any provided configuration
+   * avoiding any need for set_key() usage on the initialized object.
+   *
+   * @param config containing sm.encryption_key and sm.encryption_type for
+   *        initializing the object.
+   */
+  EncryptionKey(const Config& config);
```

Comment:
I don't mind these functions, but I don't think we need them in this PR. We should have exactly one class in the whole code base that uses the configuration key `sm.encryption_key`, and this seems like a good place to put it. But it would be better to make those changes all together.

If we use the key from the `Array` object, we shouldn't need to use this method to get keys at all in this PR.